app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -137,6 +149,108 @@ fun CreationDialog(
+                    Modifier.weight(1f)
+                        .padding(start = 16.dp, end = 16.dp, top = 8.dp)
+                        .verticalScroll(rememberScrollState()),
+                verticalArrangement = Arrangement.spacedBy(4.dp)) {
+                  if (selectedFolder == null) {
+                    userRootFolders.value.forEach { folder ->
+                      Box(
+                          modifier =
+                              Modifier.testTag("FileSystemPopupFolderChoiceBox" + folder.id)
+                                  .fillMaxWidth()
+                                  .background(
+                                      color = MaterialTheme.colorScheme.surface,
+                                      shape = RoundedCornerShape(8.dp))
+                                  .clickable {
+                                    folderViewModel.getSubFoldersOfNoStateUpdate(
+                                        folder.id,
+                                        onSuccess = { subFolders -> folderSubFolders = subFolders })
+                                    selectedFolder = folder
+                                  }
+                                  .padding(12.dp)) {
+                            Text(
+                                modifier = Modifier.testTag("FileSystemPopupFolderChoiceText"),
+                                text = folder.name,
+                                style = MaterialTheme.typography.bodyMedium,
+                                color = MaterialTheme.colorScheme.onSurface)
+                          }
+                    }
+                  } else {
+                    folderSubFolders.forEach { subFolder ->
+                      Box(
+                          modifier =
+                              Modifier.fillMaxWidth()
+                                  .background(
+                                      color = MaterialTheme.colorScheme.surface,
+                                      shape = RoundedCornerShape(8.dp))
+                                  .clickable {
+                                    folderViewModel.getSubFoldersOfNoStateUpdate(
+                                        subFolder.id,
+                                        onSuccess = { subFolders -> folderSubFolders = subFolders })
+                                    selectedFolder = subFolder
+                                  }
+                                  .padding(12.dp)) {
+                            Text(
+                                text = subFolder.name,
+                                style = MaterialTheme.typography.bodyMedium,
+                                color = MaterialTheme.colorScheme.onSurface)
+                          }
+                    }
+                  }
+                }
+
+            Box(modifier = Modifier.fillMaxWidth().padding(8.dp)) {
+              ElevatedButton(
+                  onClick = { selectedFolder = null }, modifier = Modifier.fillMaxWidth()) {
```

Comment:
Consider using the added function in the view model to set the state to null

Reply:
I thought it was better to do this like that as here we are referencing our local selectedFolder and do not need to go through the viewModel.

Reply:
True, even though I think changes in the general state should be observed in the local one